Source/WebCore/page/scrolling/mac/ScrollingTreeScrollingNodeDelegateMac.mm

```diff
@@ -294,14 +312,38 @@
 
 void ScrollingTreeScrollingNodeDelegateMac::releaseReferencesToScrollerImpsOnTheMainThread()
 {
-    if (m_verticalScrollerImp || m_horizontalScrollerImp) {
+    if (m_scrollerPair->verticalScroller().scrollerImp() || m_scrollerPair->horizontalScroller().scrollerImp()) {
         // FIXME: This is a workaround in place for the time being since NSScrollerImps cannot be deallocated
         // on a non-main thread. rdar://problem/24535055
-        WTF::callOnMainThread([verticalScrollerImp = WTFMove(m_verticalScrollerImp), horizontalScrollerImp = WTFMove(m_horizontalScrollerImp)] {
+        WTF::callOnMainThread([verticalScrollerImp = m_scrollerPair->verticalScroller().scrollerImp(), horizontalScrollerImp = m_scrollerPair->horizontalScroller().scrollerImp()] {
         });
     }
 }
 
+void ScrollingTreeScrollingNodeDelegateMac::initScrollbars()
+{
+    m_scrollerPair->init();
+}
+
+void ScrollingTreeScrollingNodeDelegateMac::updateScrollbarLayers()
+{
+    if (m_scrollerPair)
```

Comment:
Why the null test here but not elsewhere? Aren't we guaranteed that the pair is non-null?

Reply:
Removed these checks.